### DISCUSSION TITLE
TSNsched-based Gate Scheduling - Showcase

### DISCUSSION THREAD

**Discussion by adamgeorge309** [inet-framework/inet#793] [Showcases]
Discuss the "TSNsched-based Gate Scheduling" showcase here. Ideas and recommendations are welcome.

**Comment by tuantitu22** [comment]
Hello, when I run simulation, error dialogbox pops up. It says: TSNsched command execution failed, make sure TSNSCHED_ROOT is set and Microsoft Z3 is installed." Could you please give me guidance how to do it? Thanks a lot! 
By the way, I run Omnet++ in windows system.

**Reply by tuantitu22** [reply]
I have added system environment variable in windows and installed Z3, but it seems still not work.

**Comment by avarga** [comment]
The message in the dialog box is definitely misleading. You need to install the **TSNsched** tool from `https://github.com/ACassimiro/TSNsched`, make sure it works, and set the `TSNSCHED_ROOT` environment variable to point to its directory.

If you just want to run a Z3-based gate schedule configurator which is part of INET (no external dependency to be installed), run the `sat` showcase simulation next to this (`tsnsched`) one.

**Reply by tuantitu22** [reply]
But TSNsched can only be installed in linux, right?

**Reply by avarga** [reply]
> But TSNsched can only be installed in linux, right?

Not sure, they don't say it explicitly on their web page. It's mostly Java so runs anywhere, and the `libs/z3-turnkey-4.12.2.jar` file in it appears to contain binaries for multiple platforms: Linux 64-bit, macOS, Windows 32 and 64 bits. (It contains a copy of the Z3 lib, so it doesn't need to be installed on the system.) So I don't think there's any serious reason why it shouldn't run on all those platforms.

**Comment by avarga** [comment]
I refined the error message to be more explanatory: *"TSNsched command execution failed -- make sure you have the TSNsched tool from https://github.com/ACassimiro/TSNsched installed, it works, and the TSNSCHED_ROOT environment variable points to its directory. Command was: java -classpath ... ... ..."*
